### DISCUSSION TITLE
Use of CoupledPenaltyInterfaceDiffusion with vectors

### DISCUSSION THREAD

**Discussion by TLWise** [idaholab/moose#18068] [Q&A Modules: General] [answered]
I am using the CoupledPenaltyInterfaceDiffusion interface kernel with velocity which is defined as a LAGRANGE_VEC, I have defined the velocity components  (vel_x, vel_y) as AuxKernels. During execution of the code an error is displayed for the variables (velocity, vel_x, vel_y) that I am attempting to use within CoupledPenaltyInterfaceDiffusion. The error indicates that neither of these variables is of the nonlinear type expected by the Kernel. Is there a way to transform these values for use in the CoupledPenaltyInterfaceDiffusion interface kernel?

**Comment by GiudGiud** [comment, marked as answer]
Hi 

The CoupledPenaltyInterfaceDiffusion expects a non vector variable. You will have to either recode this interface kernel to use a vector variable or use 3 non linear variables for each component of velocity. 

Guillaume
